oxidation/libparsec/crates/types/src/time.rs

```diff
@@ -113,13 +114,69 @@ impl DateTime {
         now.into()
     }
 
-    /// Return a date-time formatted string in the rfc3339 format with a precision update to milliseconds.
-    /// Equivalent to ISO-8601
+    // We use RFC3339 format for some serialization (e.g. in `OrganizationFileLink`)
+
     pub fn to_rfc3339(&self) -> String {
-        self.0.to_rfc3339_opts(chrono::SecondsFormat::Micros, false)
+        self.0.to_rfc3339_opts(chrono::SecondsFormat::AutoSi, true)
+    }
+
+    pub fn from_rfc3339(s: &str) -> Result<Self, chrono::ParseError> {
+        s.parse().map(|dt: chrono::DateTime<chrono::Utc>| dt.into())
+    }
+}
+
+impl std::convert::AsRef<chrono::DateTime<chrono::Utc>> for DateTime {
+    #[inline]
+    fn as_ref(&self) -> &chrono::DateTime<chrono::Utc> {
+        &self.0
     }
 }
 
+impl std::fmt::Debug for DateTime {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("DateTime").field(&self.to_rfc3339()).finish()
+    }
+}
+
+// `FromStr` is a convenient but ambiguous shortcut, we should only use it
+// for tests and prefere `DateTime::from_rfc3339` elsewhere
+// TODO: prevent me from being used outside of test code
+impl std::str::FromStr for DateTime {
```

Comment:
```suggestion
#[cfg(test)]
impl std::str::FromStr for DateTime {
```

Reply:
nop, see https://github.com/Scille/parsec-cloud/issues/3344

Reply:
:(

Reply:
One way would be to not provide the implementation of `FromStr` even if it's more convenient for the tests

Reply:
Or we could use a combination `#[cfg(any(test, features = "mock-time"))]`